src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.Parsers.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp.Formatting
+{
+    public static partial class CSharpFormattingOptions
+    {
+        internal static bool ParseParenthesesSpaceOptions(string value, SpacingWithinParenthesesOption parenthesesSpacingOption)
+            => (from v in value.Split(',').Select(v => v.Trim())
+                let option = ConvertToSpacingOption(v)
+                where option.HasValue
+                select option.Value)
+                .Any(v => v == parenthesesSpacingOption);
+
+        private static SpacingWithinParenthesesOption? ConvertToSpacingOption(string value)
+        {
+            switch (value)
+            {
+                case "expressions": return SpacingWithinParenthesesOption.Expressions;
+                case "type_casts": return SpacingWithinParenthesesOption.TypeCasts;
+                case "control_flow_statements": return SpacingWithinParenthesesOption.ControlFlowStatements;
+                default: return null;
+            }
+        }
+
+        internal static object ParseEditorConfigSpacingAroundBinaryOperator(string binaryOperatorSpacingValue)
+        {
+            switch (binaryOperatorSpacingValue)
+            {
+                case "ignore": return BinaryOperatorSpacingOptions.Ignore;
+                case "none": return BinaryOperatorSpacingOptions.Remove;
```

Comment:
I'm preferring the names we expose in the IDE over the variable names.  It would be a breaking API change the modify the enum names as `BinaryOperatorSpacingOptions` is a public type.